### ISSUE TITLE
attempt to index '(string)' with '"gsub"'

### ISSUE THREAD

**Issue by Zash** [andremm/typedlua#106]
```
local function a(s:string):string
  return (s:lower():gsub("a","b"));
end
```

should be equivalent to `string.gsub(s:lower(), "a", "b")` and `string.gsub(string.lower(s), "a", "b")` but gives a `type error`.

**Comment by Veltas** [comment]
You're right; there appears to be a bug in the compiler for the `:` operator.

It thinks you're trying to give a 1-tuple `(string)` to `gsub`. As a workaround, it works when you wrap the expression in brackets:
```lua
(s:lower()):gsub("a", "b")
```

**Comment by Veltas** [comment]
This is a completely nonsense report, something is clearly broken here:
```lua
("a"):gsub("a", "b"):lower()
```

    attempt to index '(string, integer)' with '"lower"'
